quizbizz/src/pages/quizquestion/QuizQuestion.jsx

```diff
@@ -0,0 +1,178 @@
+import { useQuiz } from "../../context/quiz-context";
+import { useNavigate } from "react-router-dom";
+import { useToast } from "../../context/toast-context";
+import "./quizquestion.css";
+
+const QuizQuestion = () => {
+  const { quiz, setQuiz, answersSelected, setAnswersSelected } = useQuiz();
+  const navigate = useNavigate();
+  const { showToast } = useToast();
+
+  //   if (quiz === null) {
+  //     navigate("/");
+  //     return null;
+  //   }
+
+  const quizName = quiz?.quizName;
+  const quizImage = quiz?.quizImage;
+  const [questionToShow, setQuestionToShow] = useState(0);
+  const questions = quiz?.questions;
+  const question = questions?.[questionToShow];
+  const options = question?.options;
+
+  // console.log(options);
+
+  const handleNextButtonClick = () => {
+    if (questionToShow + 1 >= questions.length) {
+      return;
+    }
+    if (
+      answersSelected[questionToShow] !== "" &&
+      answersSelected[questionToShow] !== undefined
+    ) {
+      setQuestionToShow((prev) => prev + 1);
+    } else {
+      showToast("Please Select a option first", "ERROR");
+    }
+  };
+
+  const handlePrevButtonClick = () => {
+    if (questionToShow - 1 < 0) {
+      return;
+    }
+    if (answersSelected[questionToShow] !== "") {
+      setQuestionToShow((prev) => prev - 1);
+    } else {
+      showToast("Please Select a option first", "ERROR");
+    }
+  };
+
+  const handleInputChange = (index) => {
+    let tempAnswers = answersSelected;
+    tempAnswers[questionToShow] = String.fromCharCode(index + 97);
+    setAnswersSelected([...tempAnswers]);
+  };
+
+  return (
+    <div>
+      <Navbar />
+      <main className="page-container">
+        <Sidebar />
+        <div className="col-2 hero-section-container">
+          {/* <!--  ******************************** Hero Section Start ********************************  --> */}
+          <div className="quiz-heading-container">
+            <div className="wrapper">
+              <h1 className="h4">{quizName} Quiz</h1>
+              <p className="font-medium primary-text-color">
+                Read the following instructions carefully
+              </p>
+            </div>
+
+            {/* <p className="timer-container font-medium-large weight-bold">
+              Timer: 20:00 min
+            </p> */}
+          </div>
+
+          {/* <!--  ******************************** Quiz Image and question container Start ********************************  --> */}
+          <div className="quiz-info-container quiz-question-container">
+            <div className="image-container">
+              <img
+                src={quizImage}
+                className="img-responsive quiz-info-image"
+                alt={quizName}
+              />
+            </div>
+            <div className="quiz-question">
+              <h2 className="font-medium-large weight-semi-bold primary-color">
+                Question {question.questionNumber} / {questions.length}:
```

Comment:
you can use like this too and why you are not using ? .
For example : question?.questionNumber ?? 0

Reply:
Here optional chaining is not required that is why i am not  using it like that